src/components/Explore/TokenRow.tsx

```diff
@@ -0,0 +1,314 @@
+import { Trans } from '@lingui/macro'
+import CurrencyLogo from 'components/CurrencyLogo'
+import { useCurrency, useToken } from 'hooks/Tokens'
+import { TimePeriod, TokenData } from 'hooks/useTopTokens'
+import { darken } from 'polished'
+import React from 'react'
+import { ArrowDownRight, ArrowUpRight, Heart } from 'react-feather'
+import { ArrowDown, ArrowUp } from 'react-feather'
+import styled from 'styled-components/macro'
+import { formatAmount, formatDollarAmount } from 'utils/formatDollarAmt'
+
+//   @media screen and (max-width: 1225px) and (min-width: 1045px) {}
+const TokenRowWrapper = styled.div`
+  width: 100%;
+  height: 60px;
+  display: grid;
+  padding: 0px 12px;
+  grid-template-columns: 1.2fr 1fr 7fr 4fr 4fr 4fr 4fr 5fr 2fr;
+  font-size: 15px;
+  line-height: 24px;
+  margin: 4px 0px;
+  max-width: 960px;
+
+  ${({ theme }) => theme.mediaWidth.upToLarge`
+  grid-template-columns: 1.2fr 1fr 6fr 4fr 4fr 4fr 4fr 3fr;
+  gap: 10px;
+  .col-hide-1 {
+    display: none;
+  }
+  `};
+
+  ${({ theme }) => theme.mediaWidth.upToMedium`
+  grid-template-columns: 1.2fr 1fr 7fr 4fr 4fr 4fr 2.5fr;
+  width: fit-content;
+  .col-hide-1 {
+    display: none;
+  }
+  .col-hide-2 {
+    display: none;
+  }
+  `};
+
+  ${({ theme }) => theme.mediaWidth.upToSmall`
+  grid-template-columns: 1.2fr 1fr 7fr 4fr 4fr 2fr;
+  width: fit-content;
+  .col-hide-1 {
+    display: none;
+  }
+  .col-hide-2 {
+    display: none;
+  }
+  .col-hide-3 {
+    display: none;
+  }
+  `};
+
+  ${({ theme }) => theme.mediaWidth.upToExtraSmall`
+  grid-template-columns: 1fr 7fr 4fr 4fr;
+  width: fit-content;
+  .col-hide-1 {
+    display: none;
+  }
+  .col-hide-2 {
+    display: none;
+  }
+  .col-hide-3 {
+    display: none;
+  }
+  .col-hide-4 {
+    display: none;
+  }
+  `};
+`
+const HeaderRow = styled(TokenRowWrapper)`
+  width: 100%;
+  height: 48px;
+  color: ${({ theme }) => theme.text2};
+  font-size: 12px;
+  line-height: 16px;
+  border-bottom: 1px solid;
+  border-color: ${({ theme }) => theme.bg3};
+  border-radius: 8px 8px 0px 0px;
+`
+const FavoriteContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  padding: 14px 0px;
+  gap: 10px;
+  color: ${({ theme }) => theme.text2};
+
+  .fav {
+    fill: ${({ theme }) => theme.primary1};
+    color: ${({ theme }) => theme.primary1};
+  }
+`
+const ListNumberContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  padding: 14px 0px;
+  gap: 10px;
+  color: ${({ theme }) => theme.text2};
+`
+const NameContainer = styled.div`
```

Comment:
I think you can put a lot of this in a `Cell` components and then extend the specific Cells like this https://styled-components.com/docs/basics#extending-styles